contextid.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"context"
+	"net/http"
+)
+
+// AddContextID wraps a http.Handler to add a monotonically increasing
+// uint to the context of the http.Request with the key "id" (string)
+// as it passes through the request to the next handler.
+func AddContextID(next http.Handler) http.Handler {
+	ids := make(chan uint)
+	go func() {
+		for id := uint(0); ; id++ {
+			ids <- id
+		}
+	}()
```

Comment:
Added to #17 - trivial enough not to need its own issue. Not fixing now because I just refactored what was there - fixing it is outside the scope of this PR.

Reply:
Added TODO to code.